calendar-bundle/src/Resources/contao/classes/Calendar.php

```diff
@@ -593,6 +604,22 @@ public function purgeOldFeeds()
 
 		return $arrFeeds;
 	}
+
+	/**
+	 * Return the page object with loaded details for the given page ID
+	 * 
+	 * @param integer $intPageId
+	 * @return PageModel|null
+	 */
+	private function getPageWithDetails($intPageId)
+	{
+		if (isset(self::$arrPageCache[$intPageId]))
+		{
+			return self::$arrPageCache[$intPageId];
+		}
+
+		return (self::$arrPageCache[$intPageId] = PageModel::findWithDetails($intPageId));
```

Comment:
I think we do it the other way round most of the time. _(just a nitpick ☺️)_
```suggestion
		if (!isset(self::$arrPageCache[$intPageId]))
		{
			self::$arrPageCache[$intPageId] = PageModel::findWithDetails($intPageId)
		}

		return self::$arrPageCache[$intPageId];
```